### PR TITLE
Add analytics support

### PR DESCRIPTION
**Pull request by abnersajr**
#### What does this PR do?

Is related to the issue #29 
Adds Google Analytics plugin to the theme.

#### Where should the reviewer start?

Creating the example as mentioned below.

#### How should this be manually tested?

1. You should create an example project using the theme.
2. Add a Tracking Id in the config file. You can use a fake one just to check if the script is included. In case you want a fully working version, a valid tracking should be getting from GA admin.

#### Any background context you want to provide?

The plugin used is the [gatsby-plugin-google-analytics](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-google-analytics)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,23 +34,30 @@ module.exports = {
...
+| `spaceId`                | `string` | Your Contentful space ID                                   | `true`   |
+| `accessToken`            | `string` | Your Contentful content delivery API access token          | `true`   |
+| `managementToken`        | `string` | Your Contentful personal access                            | `true`   |
+| `environmentId`          | `string` | Your Contentful environment ID. (e.g. `master`)            | `true`   |
+| `mailTo`                 | `string` | The e-mail address for people apply                        | `true`   |
+| `googleAnalyticsOptions` | `object` | Your Google Analytics options                              | `false`  |
 
 Create your Contentful space, generate all the needed api keys and fill them in.
 
+📊 :Google Analytics is optional, however by default this is placed on body and you **must** pass the key `trackingId` to have the GA script included.
```
Comment (matheusazzi) [inline README.md]: @abnersajr Will the script be injected on body even if a `trackingId` isn't passed?

Comment (abnersajr) [reply README.md]: No. The plugin checks  there is a `trackingId`

File: README.md [hunk 2]
```diff
@@ -34,23 +34,30 @@ module.exports = {
         accessToken: '',
         managementToken: '',
         environmentId: '',
+        mailTo: 'become@jobs.com',
```
Comment (iagodahlem) [inline README.md]: I think this `mailTo` option could be just an empty string as well.

File: README.md [hunk 3]
```diff
@@ -34,23 +34,30 @@ module.exports = {
...
+        googleAnalyticsOptions: {...}
```
Comment (iagodahlem) [inline README.md]: As the GA is not a required thing, I think we don't need to define it here too, only options that are really required to work.

File: README.md [hunk 4]
```diff
@@ -34,23 +34,30 @@ module.exports = {
...
+| `accessToken`            | `string` | Your Contentful content delivery API access token          | `true`   |
+| `managementToken`        | `string` | Your Contentful personal access                            | `true`   |
+| `environmentId`          | `string` | Your Contentful environment ID. (e.g. `master`)            | `true`   |
+| `mailTo`                 | `string` | The e-mail address for people apply                        | `true`   |
+| `googleAnalyticsOptions` | `object` | Your Google Analytics options                              | `false`  |
 
 Create your Contentful space, generate all the needed api keys and fill them in.
 
+📊 :Google Analytics is optional, however by default this is placed on body and you **must** pass the key `trackingId` to have the GA script included.
+All options are available on this [link](https://www.gatsbyjs.org/packages/gatsby-plugin-google-analytics/#how-to-use).
```
Comment (iagodahlem) [inline README.md]: I think it's not good to use the word "link" for links. What do think about something more specific? `Check all the options available at the gatsby-plugin-google-analytics [documentation](...)`.

File: README.md (hunk 1 above)
Comment (iagodahlem) [reply README.md]: Is this `:` at the start of the sentence intentional? 🤔

Also, what do you think about this sentence instead?

```
Google Analytics is an optional feature if you really want to include analytics scripts in your site, you **must** pass the `trackingId` key to the `googleAnalyticsOptions` object.
```

File: README.md [hunk 5]
```diff
@@ -34,23 +34,30 @@ module.exports = {
...
+| Option Name              | Type     | Description                                                | Required |
+| :----------------------- | :------- | :--------------------------------------------------------- | :------- |
+| `basePath`               | `string` | The base path where your site will live. (e.g. `/careers`) | `false`  |
+| `title`                  | `string` | The main title used in the header                          | `false`  |
+| `spaceId`                | `string` | Your Contentful space ID                                   | `true`   |
+| `accessToken`            | `string` | Your Contentful content delivery API access token          | `true`   |
+| `managementToken`        | `string` | Your Contentful personal access                            | `true`   |
+| `environmentId`          | `string` | Your Contentful environment ID. (e.g. `master`)            | `true`   |
+| `mailTo`                 | `string` | The e-mail address for people apply                        | `true`   |
+| `googleAnalyticsOptions` | `object` | Your Google Analytics options                              | `false`  |
```
Comment (iagodahlem) [inline README.md]: I am not sure if we should use the word `option` in this object, it seems a little bit redundant. Any ideas @matheusazzi?

File: README.md (hunk 2 above)
Comment (iagodahlem) [reply README.md]: Actually, I don't know if these mailTo changes should be here. 🤔

File: packages/gatsby-theme-careers/gatsby-config.js [hunk 6]
```diff
@@ -18,5 +21,14 @@ module.exports = ({
         accessToken,
       },
     },
+    {
+      resolve: `gatsby-plugin-google-analytics`,
```
Comment (iagodahlem) [inline packages/gatsby-theme-careers/gatsby-config.js]: Would you mind to change this to use single quotes? I still don't know why the Gatsby community uses template string in the gatsby config file.

File: packages/gatsby-theme-careers/gatsby-config.js [hunk 7]
```diff
@@ -18,5 +21,14 @@ module.exports = ({
...
+      options: {
+        // Defines where to place the tracking script - `true` in the head and `false` in the body
```
Comment (iagodahlem) [inline packages/gatsby-theme-careers/gatsby-config.js]: I think we can remove these comments from GA plugin.

File: README.md (hunk 1 above)
Comment (abnersajr) [reply README.md]: Fixed

File: packages/gatsby-theme-careers/gatsby-config.js (hunk 7 above)
Comment (abnersajr) [reply packages/gatsby-theme-careers/gatsby-config.js]: Done

File: packages/gatsby-theme-careers/gatsby-config.js (hunk 6 above)
Comment (abnersajr) [reply packages/gatsby-theme-careers/gatsby-config.js]: Done

File: README.md (hunk 3 above)
Comment (abnersajr) [reply README.md]: Removed

File: README.md (hunk 4 above)
Comment (abnersajr) [reply README.md]: Changed

File: README.md (hunk 2 above)
Comment (iagodahlem) [reply README.md]: Still here.

File: README.md (hunk 3 above)
Comment (iagodahlem) [reply README.md]: Still here too.

File: README.md (hunk 5 above)
Comment (iagodahlem) [reply README.md]: Hey @abnersajr, I talked to @matheusazzi and we agreed that just `googleAnalytics` seems better for who will use and consume our theme. Would you mind to change?

**Pull request closed without merging**